sbi/utils/sbiutils.py

```diff
@@ -146,7 +146,12 @@ def sample_posterior_within_prior(
 
     while num_remaining > 0 and not time_over:
 
-        sample = posterior_nn.sample(num_remaining, context=x)
+        # XXX: we need this reshape here because posterior_nn.sample sometimes return
+        # leading singleton dimension instead of (num_samples), e.g., (1, 10000, 4)
+        # instead of (10000, 4). and this cant be handle by IndependentJoint.
+        sample = posterior_nn.sample(num_remaining, context=x).reshape(
```

Comment:
I made the comment here because I think it is unusual for the `.sample` method to return a leading singleton dimension, no? maybe this is a problem in how we call it or how it is initialized. I should create an issue...